app/controllers/rx_controller.rb

```diff
@@ -4,13 +4,21 @@
 class RxController < ApplicationController
   include ActionController::Serialization
 
-  skip_before_action :authenticate
+  before_action :authorize_rx
   before_action :authenticate_client
 
   protected
 
   def client
-    @client ||= Rx::Client.new(session: { user_id: ENV['MHV_USER_ID'] })
+    @client ||= Rx::Client.new(session: { user_id: mhv_correlation_id })
+  end
+
+  def authorize_rx
+    mhv_correlation_id || raise(Common::Exceptions::Unauthorized)
```

Comment:
I really do think it is unauthorized but it should have a unique minor code for unauthorized distinct from minor code related to token authorization maybe